test/kokoro/xds.sh

```diff
@@ -7,10 +7,16 @@ cd github
 
 export GOPATH="${HOME}/gopath"
 pushd grpc-go/interop/xds/client
+branch=$(git branch --all --no-color --contains "${KOKORO_GITHUB_COMMIT}" \
+    | grep -v HEAD | head -1)
+shopt -s extglob
+branch="${branch//[[:space:]]}"
```

Comment:
Why is this different from the other script?
(This is a noob question. I don't know shell that well. This might just be a better solution for the same thing.)

Reply:
I also don't know bash that well. I believe refdocs.sh strips leading and trailing whitespace only whereas this strips all whitespace, but for git branch names internal spaces aren't allowed anyways. This [SO question](https://stackoverflow.com/questions/369758/how-to-trim-whitespace-from-a-bash-variable) and particularly [this page](https://mywiki.wooledge.org/BashFAQ/073) were helpful for figuring out how this worked. I think `[[:space:]]` was slightly more description so went with that.